plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobStore.java

```diff
@@ -170,68 +122,52 @@ public void close() throws IOException {
 
     @Override
     public Map<String, Long> stats() {
-        return stats.toMap();
+        return statsMetricPublisher.getStats().toMap();
     }
 
-    public CannedAccessControlList getCannedACL() {
+    public ObjectCannedACL getCannedACL() {
         return cannedACL;
     }
 
     public StorageClass getStorageClass() {
         return storageClass;
     }
 
+    public StatsMetricPublisher getStatsMetricPublisher() {
+        return statsMetricPublisher;
+    }
+
     public static StorageClass initStorageClass(String storageClass) {
         if ((storageClass == null) || storageClass.equals("")) {
-            return StorageClass.Standard;
+            return StorageClass.STANDARD;
         }
 
-        try {
-            final StorageClass _storageClass = StorageClass.fromValue(storageClass.toUpperCase(Locale.ENGLISH));
-            if (_storageClass.equals(StorageClass.Glacier)) {
-                throw new BlobStoreException("Glacier storage class is not supported");
-            }
+        final StorageClass _storageClass = StorageClass.fromValue(storageClass.toUpperCase(Locale.ENGLISH));
```

Comment:
```suggestion
        final StorageClass storageClass = StorageClass.fromValue(storageClass.toUpperCase(Locale.ENGLISH));
```

Super nit but would be good to change that to follow naming conventions